pkg/user/tx_client_test.go

```diff
@@ -255,12 +264,36 @@ func TestEvictions(t *testing.T) {
 	require.Equal(t, seqBeforeEviction, seqAfterEviction)
 }
 
-func (suite *TxClientTestSuite) TestGasEstimation() {
+// TestSeparateGasEstimatorService ensures that if the WithEstimatorService
+// option is provided to the tx client, the separate gas estimator service is
+// used to estimate gas price and usage instead of the default connection.
+func TestSeparateGasEstimatorService(t *testing.T) {
```

Comment:
```suggestion
func TestWithEstimatorService(t *testing.T) {
```
nit to make it clearer